packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -466,17 +466,70 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
+            const overallTime = endTime - startTime;
             const content = response.content;
+            let dnstime = -1; // domainLookupEnd - domainLookupStart
+            let redirectTime = -1; // redirectEnd -redirectStart
+            let tcpHandshakeTime = -1; // connectEnd  - connectStart
+            let secureConntime = -1; // connectEnd  - secureConnectionStart
+            let responseTime = -1; // responsEnd - responseStart
+            let fetchStToRespEndTime = -1; // responseEnd  - fetchStart
+            let reqStToRespEndTime = -1; // responseEnd - requestStart
+            let networkTime = -1; // responseEnd - startTime
+            const spReqDuration = response.headers.get("sprequestduration");
+
+            const resources1 = performance.getEntriesByType("resource");
+            // Usually the latest fetch call is to the end of resources, so we start from the end.
+            for (let i = resources1.length - 1; i > 0; i--) {
+                const indResTime = resources1[i] as PerformanceResourceTiming;
+                const resource_name = indResTime.name;
+                const resource_initiatortype = indResTime.initiatorType;
+                if ((resource_initiatortype.localeCompare("fetch") === 0) && (resource_name.localeCompare(url) === 0)) {
+                        redirectTime = indResTime.redirectEnd - indResTime.redirectStart;
+                        dnstime = indResTime.domainLookupEnd - indResTime.domainLookupStart;
+                        tcpHandshakeTime = indResTime.connectEnd - indResTime.connectStart;
+                        secureConntime = (indResTime.secureConnectionStart > 0) ? (indResTime.connectEnd - indResTime.secureConnectionStart) : 0;
+                        responseTime = indResTime.responseEnd - indResTime.responseStart;
+                        fetchStToRespEndTime = (indResTime.fetchStart > 0) ? (indResTime.responseEnd - indResTime.fetchStart) : 0;
+                        reqStToRespEndTime = (indResTime.requestStart > 0) ? (indResTime.responseEnd - indResTime.requestStart) : 0;
+                        networkTime = (indResTime.startTime > 0) ? (indResTime.responseEnd - indResTime.startTime) : 0;
+                        break;
+                }
+            }
+            let clientDuration = -1;
+            if (spReqDuration) {
+                    clientDuration = parseInt(spReqDuration, 10) - responseTime;
+            }
+            const clientTime = overallTime - networkTime;
             event.end({
                 trees: content.trees?.length ?? 0,
                 blobs: content.blobs?.length ?? 0,
                 ops: content.ops?.length ?? 0,
                 headers: Object.keys(headers).length !== 0 ? true : undefined,
                 sprequestguid: response.headers.get("sprequestguid"),
                 sprequestduration: TelemetryLogger.numberFromString(response.headers.get("sprequestduration")),
+                isptionscall: isOptionsCall,
```

Comment:
wrong spelling. it should be,
{
isOptionsCall,
}

Reply:
resolved.